### PR TITLE
#529 grafana and docker compose

### PR DESCRIPTION
**Pull request by andreievg**
Fixes: #529 

For full details see screencast from telegram

### Changes

* New readme file for information about (wasn't sure if it needs to be in the docs)
* demo server folder that can be pushed to demo server (see readme files)
* allow for volumes to work in our current image (by creating a copy of fresh database initialisation and copying during image startup if one doesn't exist, i.e. if it using mounted empty volume)
* create docker compose that exposes grafana and mflow image together (linked their networks) and mounts volumes on local system
* demo server nginx config to live in demo server folder

I've updated demo server with 5002 - mFlow and 5003 - Grafana (reserving 5001 for Grafana of first instance)

Also made a of a 'release' (release candidate)

### CODE REVIEW

**Comment by nmadruga** [comment]
@CarlosNZ would be probably good to get this one also into `main`. I'll have a look now

**Comment by CarlosNZ** [comment]
> @CarlosNZ would be probably good to get this one also into `main`. I'll have a look now

Yup, have had a quick look through the video, but not too thoroughly. I think the PR itself is probably pretty solid, it's just more a matter of making sure we understand what we need to to be able to manage it correctly.

File: docker/database.sh [hunk 1]
```diff
@@ -20,3 +20,6 @@ echo '--- ADDING DATA'
 echo '--- RUNING POST INSTALL'
 ./database/turn_on_row_level_security.sh 
 ./database/post_data_insert.sh 
+
+echo '--- COPY CLEAN DATABASE TO BE USED IF NO VOLUMES ARE MOUNTED'
+cp -R /var/lib/postgresql/12/main/ ./fresh_db
```
Comment (nmadruga) [inline docker/database.sh:25]: video `mflow-grafana-1.mov` 3:55

**Review by nmadruga — APPROVED** [review]
Might be good to combine this with guide in `/documention/demo_server_guide.md`:
https://github.com/openmsupply/application-manager-server/wiki/demo_server_guide

Missing steps to be added to demo_server_guide:
- Copy or new volume of database
- Create tag on branch before creating docker image

**Comment by nmadruga** [comment]
video mflow-grafana-1.mov 10:20 - start upload on demo server

File: docker/demo_server/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    environment:
+      SMTP_PASSWORD: 'smt_password_goes_here'
+    volumes:
+      - ./app_postgres_on_port_${PORT_APP}:/var/lib/postgresql/12/main/
+      - ./app_snapshots_on_port_${PORT_APP}:/usr/src/application-manager-server/build/database/_snapshots
+    tty: true
+  grafana:
+    image: 'grafana/grafana'
+    ports:
+      - '${PORT_DASH}:3000'
```
Comment (nmadruga) [inline docker/demo_server/docker-compose.yml:20]: We can only use this on the server to test? I was trying to find a way to test this locally @andreievg

**Review by nmadruga — APPROVED** [review]

File: docker/demo_server/docker-compose.yml (hunk 2 above)
Comment (andreievg) [reply docker/demo_server/docker-compose.yml:20]: can use this locally, but have to change the docker compose and exclude `msupplyfoundation/` (or push the image to `msupplyfoundation/`) or tag the image with also `msupplyfoundation/`

**Pull request merged**